File: tests/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/dsl/GitHubDsl.java
package com.optivem.atddaccelerator.templategenerator.systemtest.dsl;

import com.optivem.atddaccelerator.templategenerator.systemtest.clients.GithubClient;
import com.optivem.atddaccelerator.templategenerator.systemtest.dsl.github.helpers.*;
import com.optivem.atddaccelerator.templategenerator.systemtest.util.Language;

public class GitHubDsl {
    private final ReadmeClient readmeClient;
    private final FileClient fileClient;
    private final WorkflowClient workflowClient;
    private final PagesClient pagesClient;
    private final DockerComposeClient dockerComposeClient;
    private final RepositoryClient repositoryClient;
    private final PackageClient packageClient;

    public GitHubDsl(GithubClient client) {
        this.readmeClient = new ReadmeClient(client);
        this.fileClient = new FileClient(client);
        this.workflowClient = new WorkflowClient(client);
        this.pagesClient = new PagesClient(client);
        this.dockerComposeClient = new DockerComposeClient(client);
        this.repositoryClient = new RepositoryClient(client);
        this.packageClient = new PackageClient(client);
    }

    public void verifyReadmeHasBadges(Language systemLanguage, Language systemTestLanguage) {
        readmeClient.verifyReadmeHasBadges(systemLanguage, systemTestLanguage);
    }

    public void verifyPathsExist(Language systemLanguage, Language systemTestLanguage) {
        fileClient.verifyPathsExist(systemLanguage, systemTestLanguage);
    }

    public void verifyWorkflowsPass(Language systemLanguage, Language systemTestLanguage) {
        workflowClient.verifyWorkflowsPass(systemLanguage, systemTestLanguage);
    }

    public void verifyPagesEnabled() {
        pagesClient.verifyPagesEnabled();
    }

    public void verifyDockerComposeImage(Language systemLanguage, Language systemTestLanguage) {
        dockerComposeClient.verifyDockerComposeImage(systemLanguage, systemTestLanguage);
    }

    public void deleteRepository() {
        repositoryClient.deleteRepository();
    }

    public void verifyRepositoryExists() {
        repositoryClient.verifyRepositoryExists();
    }

    public void verifyPackagesExist(Language systemLanguage) {
        packageClient.verifyPackagesExist(systemLanguage);
    }
}

File: console/tests/Dsl/GitHubDsl.cs
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Clients;
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Dsl.GitHub.Helpers;
using Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Dsl
{
    public class GitHubDsl
    {

        public void VerifyRepositoryExists()
        {
            _repositoryClient.VerifyRepositoryExists();
        }
        private readonly ReadmeClient _readmeClient;
        private readonly FileClient _fileClient;
        private readonly WorkflowClient _workflowClient;
        private readonly PagesClient _pagesClient;
        private readonly DockerComposeClient _dockerComposeClient;
        private readonly RepositoryClient _repositoryClient;
        private readonly PackageClient _packageClient;

        public GitHubDsl(GithubClient client)
        {
            _readmeClient = new ReadmeClient(client);
            _fileClient = new FileClient(client);
            _workflowClient = new WorkflowClient(client);
            _pagesClient = new PagesClient(client);
            _dockerComposeClient = new DockerComposeClient(client);
            _repositoryClient = new RepositoryClient(client);
            _packageClient = new PackageClient(client);
        }

        public void VerifyReadmeHasBadges(string systemLanguage, string systemTestLanguage)
        {
            _readmeClient.VerifyReadmeHasBadges(systemLanguage, systemTestLanguage);
        }

        public void VerifyPathsExist(string systemLanguage, string systemTestLanguage)
        {
            _fileClient.VerifyPathsExist(systemLanguage, systemTestLanguage);
        }

        public void VerifyWorkflowsPass(string systemLanguage, string systemTestLanguage)
        {
            _workflowClient.VerifyWorkflowsPass(systemLanguage, systemTestLanguage);
        }

        public void VerifyPagesEnabled()
        {
            _pagesClient.VerifyPagesEnabled();
        }

        public void VerifyDockerComposeImage(string systemLanguage, string systemTestLanguage)
        {
            _dockerComposeClient.VerifyDockerComposeImage(systemLanguage, systemTestLanguage);
        }

        public void VerifyPackagesExist(string systemLanguage)
        {
            _packageClient.VerifyPackagesExist(systemLanguage);
        }
    }
}